app/controllers/application_controller.rb

```diff
@@ -6,10 +6,25 @@ class ApplicationController < ActionController::API
   include Pundit
 
   before_action :configure_permitted_parameters, if: :devise_controller?
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
 
   protected
 
   def configure_permitted_parameters
-    devise_parameter_sanitizer.permit(:sign_up, keys: [:role])
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:role, :name, :real_user])
+  end
+
+  def staff_authenticate
+    authorize :staff, :staff?
+  end
```

Comment:
You are doing authorization here, not authentication, so probably `staff_authorize!` would be better  - with bang as this will raise error from pundit if user is not employee/admin.